google-cloud-spanner-hibernate-dialect/README.md

```diff
@@ -0,0 +1,7 @@
+# Google Cloud Spanner Dialect for Hibernate 5.4 ORM
+
+This project contains a Hibernate dialect for Google Cloud Spanner. This dialect is
+only compatible with Hibernate 5.4.
```

Comment:
Hmm... Not if we follow the setup that is currently in this PR. This PR assumes that we will keep the Hibernate5 and Hibernate6 dialects in different folders, and not separate them by for example maintaining separate branches. My main reason for creating separate folders instead of separate branches is that separate branches will make it less clear to users that we support both. Separate branches would also mean that any existing (external) link to the current Hibernate 5.4 dialect would no longer work as expected (it would land at the Hibernate 6 dialect).